client/keys/codec.go

```diff
@@ -4,20 +4,21 @@ import (
 	"github.com/cosmos/cosmos-sdk/codec"
 )
 
-var cdc *codec.Codec
+// KeysCdc defines codec to be used with key operations
+var KeysCdc *codec.Codec
 
 func init() {
-	cdc = codec.New()
-	codec.RegisterCrypto(cdc)
-	cdc.Seal()
+	KeysCdc = codec.New()
```

Comment:
Why do we need a public codec instance?

Reply:
because without having access to this, the new key implementation types will not be included so when trying to decode the key info using this codec while decoding and processing signed transactions it would fail otherwise. What is the ideal way to get around this? Happy to change this back to be a private codec, adding a function to register the new key types to the codec and keeping it unsealed, but I was just following what other public codecs looked like to not introduce anything new (but I think maybe this was a pattern you guys were trying to move away from)

Reply:
Got it, it's fine for me then :+1: